scripts/Units/armfido.bos

```diff
@@ -317,32 +318,6 @@ StopMoving(){
 
 #include "unit_hitbyweaponid_and_smoke.h"
 
-RequestState(requestedstate, currentstate)
```

Comment:
Does this mean the other custom on/off unit scripts don't need the `RequestState` function? I was under the impression it was necessary for on/off functionality, as it is a built-in script function defined by the engine (see https://units.tauniverse.com/tutorials/tadesign/tadesign/bosdesc.htm).

Reply:
https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/scripts/statechg.h
It seems to be defined by the game and not the engine.
I think RequestState is better used for units that have a distinct 3rd state, a "changing from state A to state B" state. armfido switch is instant, so RequestState is unnecessary.